### PR TITLE
Hotfix/Signin fetching user

### PR DESCRIPTION
**Pull request by SaturneV**
### Summary

This Pull Request fixes bugs related to authentication and user creation and fecthing. It allows to fetch the correct user from the database, or to create one if necessary when authenticating via the `SignIn` UI. It also sets the user as the current user, making it easier to access in the rest of the application's features. 
 
Once merged, this Pull Request will close issues #87 and #89.

### Changes

- Implemented `UserUtils` file and `createNewUserFromFirebaseUser` function.
- Modified `SignIn` authentification to call `getOrAddUserByUid` from User MVVM.
- Used `LaunchedEffect` to wait for the User MVVM to set the current user before navigating to the map.
- Deleted functions in `SignIn` that are no longer used due to the new implementation.

### Testing

- Implemented `ParkUtils` test functions.
- Deleted no longer relevant `SignInTest` unit tests file.

### CODE REVIEW

**Review by misterM125 — CHANGES_REQUESTED** [review]
Great PR, it seems that user fetching functionalities work properly, I just might have some advice to let you test your added functions in the signIn

File: app/src/main/java/com/android/streetworkapp/model/user/UserUtils.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.google.firebase.auth.FirebaseUser
+
+/**
+ * Creates a new [User] object from the provided [FirebaseUser].
+ *
+ * @param firebaseUser The FirebaseUser object to create the User object from.
+ * @return The User object created from the FirebaseUser.
+ */
+fun createNewUserFromFirebaseUser(firebaseUser: FirebaseUser): User {
```
[the remark this answers is not in the document]
Comment (SaturneV) [reply app/src/main/java/com/android/streetworkapp/model/user/UserUtils.kt]: Indeed, it was only useful for the file itself, so there was no need to make it a global helper function, it's refactored!

File: app/src/main/java/com/android/streetworkapp/ui/authentication/SignIn.kt [hunk 2]
```diff
@@ -49,18 +52,27 @@ fun SignInScreen(navigationActions: NavigationActions, userViewModel: UserViewMo
...
             Toast.makeText(context, "Login failed! : $it", Toast.LENGTH_LONG).show()
           })
 
+  // Wait for the user to be fetched from database before setting it
+  // as the current user and navigating to the map screen
+  LaunchedEffect(user) {
+    user?.let {
+      userViewModel.setCurrentUser(it)
+      Toast.makeText(context, "Login successful!", Toast.LENGTH_LONG).show()
+      navigationActions.navigateTo(Screen.MAP)
```
[the remark this answers is not in the document]
Comment (SaturneV) [reply app/src/main/java/com/android/streetworkapp/ui/authentication/SignIn.kt:73]: To test these lines I don't need to deal with authentication (which greatly simplifies the task!) unfortunately I'm supposed to change the value of `_user` in the ViewModel during testing to trigger the `LaunchedEffect` and I haven't managed to do so. Here again, if you have any suggestions on how I could do it, don't hesitate to let me know.

File: app/src/main/java/com/android/streetworkapp/ui/authentication/SignIn.kt [hunk 3]
```diff
@@ -49,18 +52,27 @@ fun SignInScreen(navigationActions: NavigationActions, userViewModel: UserViewMo
...
             firebaseUser = result.user
-            checkAndAddUser(firebaseUser, userViewModel)
-            firebaseUser?.let { firebaseUser -> userViewModel.getUserByUid(firebaseUser.uid) }
-            Toast.makeText(context, "Login successful!", Toast.LENGTH_LONG).show()
-            navigationActions.navigateTo(Screen.MAP)
-            userViewModel.getUserByUid(firebaseUser?.uid ?: "")
+            firebaseUser?.let { firebaseUser ->
+              userViewModel.getOrAddUserByUid(
+                  firebaseUser.uid, createNewUserFromFirebaseUser(firebaseUser))
+            }
```
[the remark this answers is not in the document]
Comment (SaturneV) [reply app/src/main/java/com/android/streetworkapp/ui/authentication/SignIn.kt:59]: Thank you for the suggestion, unfortunately even with this implementation I can't test these lines - before I can mock the viewmodel I have to successfully mock the authentication i.e. authService and despite several attempts I don't think this is something testable or mockable. If you have any ideas on how to do this, don't hesitate to let me know.

**Review by misterM125 — APPROVED** [review]

**Pull request merged**